tracee-rules/signature.go

```diff
@@ -45,63 +44,62 @@ func getSignatures(rulesDir string, rules []string) ([]types.Signature, error) {
 }
 
 func findGoSigs(dir string) ([]types.Signature, error) {
-	files, err := ioutil.ReadDir(dir)
-	if err != nil {
-		return nil, fmt.Errorf("error reading plugins directory %s: %v", dir, err)
-	}
 	var res []types.Signature
-	for _, file := range files {
-		if filepath.Ext(file.Name()) != ".so" {
-			continue
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			log.Printf("error opening plugin %s: %v", info.Name(), err)
+			return filepath.SkipDir
+		}
+		if filepath.Ext(info.Name()) != ".so" || info.IsDir() == true {
+			return filepath.SkipDir
 		}
-		p, err := plugin.Open(filepath.Join(dir, file.Name()))
+		p, err := plugin.Open(filepath.Join(path, info.Name()))
 		if err != nil {
-			log.Printf("error opening plugin %s: %v", file.Name(), err)
-			continue
+			log.Printf("error opening plugin %s: %v", info.Name(), err)
+			return filepath.SkipDir
 		}
 		export, err := p.Lookup("ExportedSignatures")
 		if err != nil {
-			log.Printf("missing Export symbol in plugin %s", file.Name())
-			continue
+			log.Printf("missing Export symbol in plugin %s", info.Name())
+			return filepath.SkipDir
 		}
 		sigs := *export.(*[]types.Signature)
 		res = append(res, sigs...)
+		return nil
+	})
+	if err != nil {
+		return nil, err
 	}
 	return res, nil
 }
 
 func findRegoSigs(dir string) ([]types.Signature, error) {
-	files, err := ioutil.ReadDir(dir)
-	if err != nil {
-		return nil, fmt.Errorf("error reading plugins directory %s: %v", dir, err)
-	}
-
-	var res []types.Signature
-
 	regoHelpersCode, err := ioutil.ReadFile(filepath.Join(dir, "helpers.rego"))
 	if err != nil {
 		log.Printf("error reading file %s/%s: %v", dir, "helpers.rego", err)
-		return res, err
+		return nil, err
 	}
 
-	for _, file := range files {
-		if filepath.Ext(file.Name()) != ".rego" {
-			continue
-		}
-		if file.Name() == "helpers.rego" {
-			continue
+	var res []types.Signature
+	err = filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if filepath.Ext(info.Name()) != ".rego" || info.IsDir() == true || info.Name() == "helpers.rego" {
+			return filepath.SkipDir
 		}
```

Comment:
This isn't an error but rather a `return nil` instead. Returning an error here will completely skip this directory. I found this out by writing a simple test:

```
func TestFindRegoSigs(t *testing.T) {
	actual, err := findRegoSigs("signatures/rego")
	require.NoError(t, err)
	assert.NotEmpty(t, actual)
}
```